### PR TITLE
remove group requests from inactivate users

### LINKED ISSUE #4456: Pending Group Request from Inactive Account
**Description of the bug**

Below is a summary of the issue and discussion from an email between Dave T, Tony C, and Martin S.

(Dave)
I clicked on the user name [of a pending request in the HS group 156 landing page] to see more, which I do before considering accepting and get “Sorry an error occurred”. I then logged in as admin and see that the account is not active.  Did someone block this account, or did the group join request somehow get sent without the account being activated.
 
In any event, we may need to work out something sensible to do to avoid the “Sorry an error occurred” in a situation like this.  A state like this is a bug, albeit perhaps not a high priority one.

(Tony)
I’m not sure how someone could’ve requested to join a HydroShare group without activating their account. That is, unless someone deactivated their account after they made the group request. I have not made any changes/modifications to user accounts in the HydroShare Admin interface. I’m going to move this conversation into GitHub so we can include some of the devs in our discussion.

Steps to reproduce the bug:
1. Go to https://www.hydroshare.org/group/156
2. Select Members/Pending
3. Click on Maryam Bagheri to view the user profile
4. 
image

### LINKED ISSUE #4484: Cannot redeem group memebership requests with inactive users
**Description of the bug**
When a user requests access to a group and then deletes (inactivates) their account, the group membership request cannot be redeemed.

Steps to reproduce the bug:
1. Create a test user
2. Request to join a group with the test user
3. Deactivate the test user account
4. As a group owner, attempt to accept/reject the group membership request from the test user
5. Observe the request reappears after a refresh

**Expected behavior**
Requests from inactive users should be removed.

**Additional information**
Here is the method which fails to redeem a user request from an inactive user.  https://github.com/hydroshare/hydroshare/blob/develop/hs_access_control/models/user.py#L151

I believe this request would get to this method call which would raise a PermissionDenied exception as the user_to_join_group is inactive.  https://github.com/hydroshare/hydroshare/blob/develop/hs_access_control/models/user.py#L205

If the user is inactive, the GroupMembershipRequest.redeemed field should be set to True and saved without any group/user permission changes.

### PR DESCRIPTION
**Pull request by devincowan**
### Issues addressed:
resolves #4456
fixes #4484

### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [ ] Automated Testing

- [ ] Sufficient User and Developer Documentation

- [x] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. Create a test user
2. Request to join a group with the test user
3. Deactivate the test user account
4. As a group owner, navigate to accept/reject the group membership request from the test user
5. Observe the request no longer appears in the list for approval

### CODE REVIEW

**Comment by devincowan** [comment]
added a test but realized that this only handles case where user deactivates their account from their Profile
doesn't handle case where admin uses Mezzanine dashboard to set `active=False`

File: hs_core/tasks.py [hunk 1]
```diff
@@ -692,6 +692,15 @@ def monthly_group_membership_requests_cleanup():
     GroupMembershipRequest.objects.filter(my_date__lte=two_months_ago).delete()
 
 
+@periodic_task(ignore_result=True, run_every=crontab(minute=30, hour=0))
```
[the remark this answers is not in the document]
Comment (devincowan) [reply hs_core/tasks.py:695]: oh I was going for once per day at 00:30

[the remark this answers is not in the document]
Comment (sblack-usu) [reply hs_core/tasks.py:695]: Maybe that is what it is doing then.  I always have to reference cron documentation when writing them.

[the remark this answers is not in the document]
Comment (devincowan) [reply hs_core/tasks.py:695]: Me too!
https://crontab.guru/#30_0_*_*_*

File: hs_access_control/models/user.py [hunk 2]
```diff
@@ -208,6 +209,46 @@ def act_on_group_membership_request(self, this_request, accept_request=True):
         this_request.redeemed = True
         this_request.save()
 
+    def redeem_group_membership_request(self, this_request, for_user=None):
```
Comment (sblack-usu) [inline hs_access_control/models/user.py:212]: I don't think this method is needed?  It seems to be mostly a duplicate of act_on_group_membership_request above.  The for_user parameter is checked and assigned after the check but never used.

Comment (devincowan) [reply hs_access_control/models/user.py]: fixed this thanks Scott

**Pull request merged**